client/state/data-layer/wpcom/sites/hosting/sftp-user.js

```diff
@@ -0,0 +1,90 @@
+/**
+ * Internal dependencies
+ */
+import { http } from 'state/data-layer/wpcom-http/actions';
+import { dispatchRequest } from 'state/data-layer/wpcom-http/utils';
+import { registerHandlers } from 'state/data-layer/handler-registry';
+import {
+	HOSTING_REQUEST_SFTP_USER,
+	HOSTING_CREATE_SFTP_USER,
+	HOSTING_RESET_SFTP_PASSWORD,
+} from 'state/action-types';
+import { errorNotice } from 'state/notices/actions';
+import { translate } from 'i18n-calypso';
+import { receiveAtomicSFTPUser, receiveAtomicSFTPUserError } from 'state/hosting/actions.js';
+
+const requestAtomicSFTPUser = action => {
+	return http(
+		{
+			method: 'GET',
+			path: `/sites/${ action.siteId }/hosting/ssh-user`,
+			apiNamespace: 'wpcom/v2',
+		},
+		action
+	);
+};
+
+const createAtomicSFTPUser = action => {
+	return http(
+		{
+			method: 'POST',
+			path: `/sites/${ action.siteId }/hosting/ssh-user`,
+			apiNamespace: 'wpcom/v2',
+			body: {},
+		},
+		action
+	);
+};
+
+const resetAtomicSFTPPassword = action => {
+	return http(
+		{
+			method: 'POST',
+			path: `/sites/${ action.siteId }/hosting/ssh-user/reset-password`,
+			apiNamespace: 'wpcom/v2',
+			body: {},
+		},
+		action
+	);
+};
+
+const receiveAtomicSFTPUserSuccess = ( action, response ) =>
+	receiveAtomicSFTPUser( action.siteId, action.userId, response );
+
+const sFTPUserError = ( { siteId, userId } ) => dispatch => {
+	dispatch( receiveAtomicSFTPUserError( siteId, userId ) );
```

Comment:
Similarly this would be a user update or remove action, the app doesn't need to know that it was part of an API error.